core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -325,7 +329,7 @@ public void deleteSnapshot(SnapshotId snapshotId) {
                 BlobPath indexPath = basePath().add("indices").add(index);
                 BlobContainer indexMetaDataBlobContainer = blobStore().blobContainer(indexPath);
                 try {
-                    indexMetaDataBlobContainer.deleteBlob(blobName);
+                    indexMetaDataFormat(snapshot.version()).delete(indexMetaDataBlobContainer, snapshotId.getSnapshot());
```

Comment:
When we deal with indices here, we know the version of the snapshot that created these indices. So `indexMetaDataFormat(snapshot.version())` here gives us the proper format (`indexMetaDataLegacyFormat` for old snapshots or `indexMetaDataFormat` for new snapshots) that we delete.